File: httprequest/upload_file/http_upload_client.py
#!/usr/bin/env python3

import requests
import os
import sys

# 指定服务器URL和要上传的文件路径
server_url = 'http://127.0.0.1:8000/'

testResponse = requests.get(server_url)
if testResponse.status_code == 200:
    print("与服务器的连接正常！")
else:
    print("无法连接到服务器！")

print("testResponse: {}".format(testResponse.text))

file_path = './test/file.bin'
with open(file_path, 'rb') as file:
    file_data = file.read()

response = requests.post(server_url, data=file_data)

if response.status_code == 200:
    print("文件上传成功！")
else:
    print("文件上传失败！")

File: httprequest/upload_file/http_upload_client.cpp
#include "HTTPRequest.hpp"
#include <cstdlib>
#include <iostream>
#include <string>
#include <fstream>
#include <iterator>

int main() {
    std::string server_url = "http://127.0.0.1:8000/";

    try
    {
        // you can pass http::InternetProtocol::V6 to Request to make an IPv6 request
        http::Request request{server_url};

        std::cout << "与服务器的连接正常！\n";

        // send a get request
        const auto testResponse = request.send("GET");
        std::cout << "testResponse: " << std::string{testResponse.body.begin(), testResponse.body.end()} << '\n'; // print the result
    }
    catch (const std::exception& e)
    {
        std::cerr << "无法连接到服务器！\n";
        std::cerr << "Request failed, error: " << e.what() << '\n';
    }

    try
    {
        std::string file_path = "./test/file.bin";
        std::ifstream file(file_path);
        if (!file) {
            std::cerr << "open file: " << file_path << " failed!\n";
            exit(1);
        }
        std::string file_data(std::istreambuf_iterator<char>{file}, {});
        http::Request request{server_url};
        const auto response = request.send("POST", file_data, {
                {"Content-Type", "text/plain"}
                });
        std::cout << std::string{response.body.begin(), response.body.end()} << '\n'; // print the result
    }
    catch (const std::exception& e)
    {
        std::cerr << "Request failed, error: " << e.what() << '\n';
    }

}

